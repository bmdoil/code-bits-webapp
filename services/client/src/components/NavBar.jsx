import React from 'react';
import { Link } from 'react-router-dom';

const NavBar = (props) => (
    <nav className="navbar is-dark" role="navigation" aria-label="main navigation">
        <section className="container">
            <div className="navbar-brand">
                <strong className="navbar-item">{props.title}</strong>
                <span
                className="nav-toggle navbar-burger"
                onClick={() => {
                    let toggle = document.querySelector(".nav-toggle");
                    let menu = document.querySelector(".navbar-menu");
                    toggle.classList.toggle("is-active"); menu.classList.toggle("is-active");
                }}>
                <span></span>
                <span></span>
                <span></span>
                </span>
            </div>
            <div className="navbar-menu">
                <div className="navbar-start">
                    <Link to="/" className="navbar-item">Home</Link>
                    {props.isAuthenticated &&
                        <Link to="/all-users" className="navbar-item">All Users</Link>
                    }
                    {props.isAuthenticated &&
                        <Link to="/all-questions" className="navbar-item">All Questions</Link>
                    }
                    {props.isAuthenticated &&
                        <Link to="/all-scores" className="navbar-item">All Scores</Link>
                    }
                    <a href="/swagger" className="navbar-item">API</a>
                </div>            
                <div className="navbar-end">
                    {!props.isAuthenticated &&
                        <Link to="/register" className="navbar-item">Register</Link>
                    }
                    {!props.isAuthenticated &&
                        <Link to="/login" className="navbar-item">Login</Link>
                    }
                    {props.isAuthenticated &&
                        <Link to="/me" className="navbar-item">My Profile</Link>
                    }
                    {props.isAuthenticated &&
                        <Link to="/signout" className="navbar-item">Signout</Link>
                    }                    
                </div>
            </div>
        </section>
    </nav>
)



export default NavBar;