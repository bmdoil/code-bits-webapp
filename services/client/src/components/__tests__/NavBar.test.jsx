import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
// https://reacttraining.com/react-router/core/api/MemoryRouter
import { MemoryRouter as Router } from 'react-router-dom'

import NavBar from '../NavBar';

const title = "Code Bites";

test('Navbar renders', () => {
    const wrapper = shallow(<NavBar title={title}/>);
    const element = wrapper.find('strong');
    expect(element.length).toBe(1);
    expect(element.get(0).props.children).toBe(title);
})
// Renders a navbar with title inside a Router object
test('Navbar snapshot renders', () => {
    const tree = renderer.create(
        <Router location="/"><NavBar title={title}/></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})