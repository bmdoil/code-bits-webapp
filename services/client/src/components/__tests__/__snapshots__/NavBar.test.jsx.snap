// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot renders 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-dark"
  role="navigation"
>
  <section
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <strong
        className="navbar-item"
      >
        Code Bites
      </strong>
      <span
        className="nav-toggle navbar-burger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </span>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-start"
      >
        <a
          className="navbar-item"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="navbar-item"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="navbar-item"
          href="/me"
          onClick={[Function]}
        >
          Profile
        </a>
      </div>
      <div
        className="navbar-end"
      >
        <a
          className="navbar-item"
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
        <a
          className="navbar-item"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          className="navbar-item"
          href="/logout"
          onClick={[Function]}
        >
          Logout
        </a>
      </div>
    </div>
  </section>
</nav>
`;
